inc/Engine/Preload/Controller/PreloadUrl.php

```diff
@@ -136,10 +144,30 @@ public function preload_url( string $url ) {
 				return;
 			}
 
+			if ( $check_duration ) {
+				$start = microtime( true );
+			}
+
 			wp_safe_remote_get(
 				user_trailingslashit( $request['url'] ),
 				$headers
 			);
+
+			if ( $check_duration ) {
+				$duration = ( microtime( true ) - $start ); // Duration of the request.
+
+				// Update average duration.
+				$previous_request_durations = get_transient( 'rocket_preload_previous_request_durations' ) ?? 0;
+				if ( $previous_request_durations <= 0 ) {
+					$previous_request_durations = $duration;
+				} else {
+					$previous_request_durations = $previous_request_durations * 0.7 + $duration * 0.3;
+				}
+				set_transient( 'rocket_preload_previous_request_durations', $previous_request_durations, 5 * 60 );
+
+				set_transient( 'rocket_preload_check_duration', $duration, 60 ); // Don't check request duration for 1 minute.
+				$check_duration = false;
+			}
```

Comment:
@jeawhanlee We can use a return here: this whole part of the code is in a for loop and we should never skip iterations of this loop.
Instead of return, we could consider using `continue`. However, we should be careful: the usleep should not be done before `$duration = ( microtime( true ) - $start ); // Duration of the request.`, otherwise, it breaks the timing computation.

So we could eventually have the following:

```
			if ( $check_duration ) {
				$duration = ( microtime( true ) - $start ); // Duration of the request.
                         }
			/**
			 * Filter the delay between each preload request.
			 *
			 * @param float $delay_between the defined delay.
			 * @returns float
			 */
			$delay_between = apply_filters( 'rocket_preload_delay_between_requests', 500000 );

			usleep( $delay_between );
                         if ( ! $check_duration ) {
				continue;
                         }
			// Update average duration.
			$previous_request_durations = get_transient( 'rocket_preload_previous_request_durations' ) ?? 0;
			if ( $previous_request_durations <= 0 ) {
				$previous_request_durations = $duration;
			} else {
				$previous_request_durations = $previous_request_durations * 0.7 + $duration * 0.3;
			}
			set_transient( 'rocket_preload_previous_request_durations', $previous_request_durations, 5 * 60 );

			set_transient( 'rocket_preload_check_duration', $duration, 60 ); // Don't check request duration for 1 minute.
			$check_duration = false;

		}
```
I don't have strong feeling for the current version or this one :)